test/HooglePlus/FilterTestSpec.hs

```diff
@@ -31,10 +43,22 @@ spec =
     result <- runTest ["Data.Maybe"] "a -> a" "\\x -> fromJust Nothing"
     result `shouldBe` False
 
-  it "Fail on invalid function 1" $ do
+  it "Fail on invalid function 2" $ do
     result <- runTest ["Data.List"] "a -> a" "\\x -> head []"
     result `shouldBe` False
 
-  it "Fail on invalid function 2" $ do
+  it "Fail on invalid function 3" $ do
     result <- runTest ["Data.List"] "a -> (a, a)" "\\x -> (head [x], last [])"
-    result `shouldBe` False
\ No newline at end of file
+    result `shouldBe` False
+
+  it "Fail on invalid function 4" $ do
+    result <- runTest ["Data.List"] "a -> (a, a)" "\\x -> (head [x], last [])"
+    result `shouldBe` False
+
+  it "Fail on invalid function 5" $ do
+    result <- runTest ["Data.Maybe"] "Either a b -> Maybe a" "\\x -> fromJust Nothing"
+    result `shouldBe` False
+  
+  it "Pass w/ type class 1" $ do
+    result <- runTest [] "(Show a, Show b) => Either a b -> String" "\\x -> show x"
+    result `shouldBe` True
```

Comment:
There are quite a few of these tests will all the same structure. Perhaps you could create a variable `testCases :: ([String],String, Bool)` to represent modules, inputs, and outputs. Then you can use something like:
```
evalCase :: ([String], String, Bool) -> IO ()
evalCase (modules, input, output) = do
   result <- runTest modules input
   when result /= output (do
    printf "failed: %s, %s => %s" modules input (show output)
   )
   result `shouldBe` output
  
...
mapM evalCase testCases